File: settlement/room/infra/hauler/Crate.java
package settlement.room.infra.hauler;

import settlement.main.SETT;
import settlement.room.main.RoomInstance;
import settlement.room.main.job.StorageCrate;

final class Crate extends StorageCrate{

	static final int size = 80;
	protected final ROOM_HAULER b;
	HaulerInstance ins;
	
	Crate(ROOM_HAULER b) {
		this.b = b;
	}

	@Override
	protected boolean is(int tx, int ty) {
		if (b.is(tx, ty)) {
			ins = b.getter.get(tx, ty);
			if (SETT.ROOMS().fData.tileData.is(tx, ty, 1)) {
				return true;
			}
		}
		return false;
	}
	
	@Override
	public boolean isStorage() {
		return true;
	}
	
	@Override
	public boolean isPrio() {
		return ins.prio();
	}
	
	@Override
	protected void count(int delta) {
		b.tally.report(this, ins, delta);
		ins.updateMasks();
	}
	
	@Override
	public boolean isFindable() {
		return !ins.storing();
	}

	@Override
	protected int max(RoomInstance ins) {
		return size;
	}

	@Override
	protected double spoilRate(RoomInstance ins) {
		return 1.0;
	}

}

File: LegacySource/settlement/room/infra/hauler/Crate.cs
using settlement.main;
using settlement.room.main;
using settlement.room.main.job;

namespace settlement.room.infra.hauler
{
    internal class Crate : StorageCrate
    {
        public static readonly int size = 80;
        private readonly ROOM_HAULER b;
        private HaulerInstance ins;

        public Crate(ROOM_HAULER b)
        {
            this.b = b;
        }

        protected override bool Is(int tx, int ty)
        {
            if (b.Is(tx, ty))
            {
                ins = b.Getter.Get(tx, ty);
                if (SETT.ROOMS().FData.TileData.Is(tx, ty, 1))
                {
                    return true;
                }
            }
            return false;
        }

        public override bool IsStorage()
        {
            return true;
        }

        public override bool IsPrio()
        {
            return ins.Prio();
        }

        protected override void Count(int delta)
        {
            b.Tally.Report(this, ins, delta);
            ins.UpdateMasks();
        }

        public override bool IsFindable()
        {
            return !ins.Storing();
        }

        protected override int Max(RoomInstance ins)
        {
            return size;
        }

        protected override double SpoilRate(RoomInstance ins)
        {
            return 1.0;
        }
    }
}